### PR TITLE
Feature/pr delete live tests (alternative to #154)

### PR DESCRIPTION
**Pull request by martinda**
This is to address issue #153. It is an alternative to PR #154 .

The benefits of being able to delete pull-requests from the REST API are:
* Higher level applications can delete pull-requests from the REST API.
* Ability to write tests for higher level applications that manipulate pull-requests via the REST API.

**Tests**
The following mock and live tests are provided in this PR:
* Attempt to delete a merged pull-request
* Attempt to delete a pull-request with an invalid pull-request version
* Attempt to delete a pull-request with an invalid pull-request id
* Delete a pull-request
 
**Complications**
A very minor complication is that the pull-request deletion live tests are into two classes. I use `PullRequestApiLiveTest` to test the deletion of a merged pull request (since that test merges a pull-request already), and the rest of the deletion scenarios are in `PullRequestDeleteApiLiveTest`.

@cdancy, your review would be greatly appreciated. Thanks and no rush.

### CODE REVIEW

File: src/test/java/com/cdancy/bitbucket/rest/features/PullRequestDeleteApiLiveTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    @AfterClass
+    public void fin() throws IOException {
+        TestUtilities.terminateGeneratedTestContents(this.api, generatedTestContents);
+    }
+
+    private PullRequestApi api() {
+        return api.pullRequestApi();
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (martinda) [reply src/test/java/com/cdancy/bitbucket/rest/features/PullRequestDeleteApiLiveTest.java]: The reason for going into another class is that the pull-request used for testing has been merged by another test, hence it cannot be deleted. But I am working on a third implementation that keeps everything in the same class.

File: src/main/java/com/cdancy/bitbucket/rest/features/PullRequestApi.java [hunk 2]
```diff
@@ -103,6 +104,18 @@ PullRequest create(@PathParam("project") String project,
...
+    @Documentation({"https://docs.atlassian.com/bitbucket-server/rest/5.13.0/bitbucket-rest.html?utm_source=%2Fstatic%2Frest%2Fbitbucket-server%2Flatest%2Fbitbucket-rest.html&utm_medium=301#idm46209337261168"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/{project}/repos/{repo}/pull-requests/{pullRequestId}")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @DELETE
+    RequestStatus delete(@PathParam("project") String project,
+                       @PathParam("repo") String repo,
+                       @PathParam("pullRequestId") long pullRequestId,
+                       @BinderParam(BindToJsonPayload.class) DeletePullRequest deletePullRequest);
```
[the remark this answers is not in the document]
Comment (martinda) [reply src/main/java/com/cdancy/bitbucket/rest/features/PullRequestApi.java]: Really nice. I will use this.

**Comment by martinda** [comment]
Ok, so I think I got a reasonable solution here. All the tests are in the same class now. I had to create another pull-request after the merge test so that I could test the deletion with something that could be deleted. I had to refactored a bit of the test setup and teardown. I also use `PayloadParam`as suggested.

**Comment by martinda** [comment]
@cdancy Are the tests successful on your side?

**Comment by cdancy** [comment]
@martinda sorry been busy with stuff :)

This all works as expected. Thanks a ton for not only cleaning this up but making the process that much better. Really ... I appreciate it.

**Pull request merged**